### Introdução
A base de dados disponibilizada tem como base o incidente do Titanic, seus dados são compostos por uma ID para cada passageiro, status de sobrevivência em valor booleano (0 - morto e 1 - vivo), tipo de passagem, nome, sexo biológico, idade, ticket, valor pago, cabine e tipo de embarque.
Obs: os dados "SibSp" e "Parch" são foram possiveís de serem identificados.

Dados observados:

-  Número total de passageiros:
    - Mortos 
    - Sobreviventes
- Quantidade em cada classe:
    - Sexo Biológico por classe
    - Sobreviventes por classe
- Passageiros por gênero:
    - Média de vivos e mortos por genero
- Passageiros média de idade:
    - Média de vivos e mortos por idade
- Há familiares?
- Media de irmãos dos passageiros
- Valor medio da passagem
- Tipo de embarque x


### Processamento

In [207]:
# Bibliotecas

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


ModuleNotFoundError: No module named 'matplotlib'

In [ ]:
# Dataset 
df = pd.read_csv("titanic.csv")
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [ ]:
# Tradução de colunas
df.rename(columns={'PassengerId': 'ID Passageiro', 'Survived': 'Sobrevivente', 'Pclass': 'Classe', 'Name': 'Nome/Sobrenome', 'Sex': 'Sexo Biológico', 'Age': 'Idade', 'SibSp': 'Num_Irmaos', 'Parch': 'Num_Pais', 'Ticket': 'Passagem', 'Fare': 'Valor da Passagem', 'Cabin': 'Cabine', 'Embarked': "Embarque"}, inplace = True)

In [ ]:
# Atualização de nome F e M
df['Sexo Biológico'] = df['Sexo Biológico'].replace({'male': 'M', 'female': 'F'})

In [ ]:
# Verifica se há valores nulos
print("Valores nulos por coluna do dataframe:")
print(df.isnull().sum())

Valores nulos por coluna do dataframe:
ID Passageiro          0
Sobrevivente           0
Classe                 0
Nome/Sobrenome         0
Sexo Biológico         0
Idade                177
Num_Irmaos             0
Num_Pais               0
Passagem               0
Valor da Passagem      0
Cabine               687
Embarque               2
dtype: int64


In [ ]:
# Preenche valores nulos da coluna Idade, Cabine e Embarque
df['Idade'] = df['Idade'].fillna(0)
df['Cabine'] = df['Cabine'].fillna('N/A')
df['Embrque'] = df['Embarque'].fillna('N/A')

In [ ]:
# Número total de passageiros
df_passageiros = df[['Sobrevivente']].value_counts()
df_passageiros

Sobrevivente
0               549
1               342
dtype: int64

In [ ]:
df_classe = df.groupby('Classe')

In [ ]:
# Quantidade de passageiros em cada classe por sexo biológico
passageiros_sexo = df_classe['Sexo Biológico'].value_counts()
passageiros_sexo

Classe  Sexo Biológico
1       M                 122
        F                  94
2       M                 108
        F                  76
3       M                 347
        F                 144
Name: Sexo Biológico, dtype: int64

In [ ]:
# Sobreviventes por classe
sobrevivente_classe = df_classe['Sobrevivente'].value_counts()
sobrevivente_classe

Classe  Sobrevivente
1       1               136
        0                80
2       0                97
        1                87
3       0               372
        1               119
Name: Sobrevivente, dtype: int64

In [ ]:
# Agrupamento por sexo
df_genero = df.groupby('Sexo Biológico')

In [ ]:
#Passageiros por gênero:
# Contagem de vivos e mortos por gênero
sobrevivente_genero = df_genero['Sobrevivente'].value_counts()
sobrevivente_genero


Sexo Biológico  Sobrevivente
F               1               233
                0                81
M               0               468
                1               109
Name: Sobrevivente, dtype: int64

In [ ]:
# Média de vivos e mortos por gênero
media_sobrev = df_genero['Sobrevivente'].mean()
media_sobrev

Sexo Biológico
F    0.742038
M    0.188908
Name: Sobrevivente, dtype: float64

In [ ]:
# Média de idade por sexo biológico
media_sexIdade = df_genero['Idade'].mean()
media_sexIdade

Sexo Biológico
F    23.203822
M    24.123345
Name: Idade, dtype: float64

In [ ]:
# Agrupar por número de irmãos
df_irmaos = df.groupby('Num_Irmaos')

In [ ]:
# Média de irmãos 
media_irmaos = df_irmaos['ID Passageiro'].mean()
media_irmaos

Num_Irmaos
0    455.370066
1    439.727273
2    412.428571
3    321.562500
4    381.611111
5    336.800000
8    481.714286
Name: ID Passageiro, dtype: float64

In [ ]:
media_passagem = df['Valor da Passagem'].mean()
media_passagem

32.204207968574636

In [200]:
#Agrupar tipos de embarque
df_embarque = df[['Embarque']].value_counts()

Embarque
S           644
C           168
Q            77
dtype: int64

### Visualização dos Dados

In [202]:
passageiros_sexo = df_classe['Sexo Biológico'].value_counts()
passageiros_sexo.plot(kind='bar');

plt.xlabel('Passageiro')
plt.ylabel('Quantidade')
plt.title('Número de Passageiros por Sexo Biológico')
plt.show()


ImportError: matplotlib is required for plotting when the default backend "matplotlib" is selected.

In [ ]:
sobrevivente_genero.plot(kind='barh');

In [ ]:
sobrevivente_classe.plot(kind='barh');

In [ ]:
df['Idade'].plot(kind='hist')

In [ ]:
df_embarque.plot(kind='barh');

In [ ]:
contagem_irmaos = df['Num_Irmao'].value_counts()
contagem_irmaos.plot(kind='bar');

plt.xlabel('Quantidade de irmãos')
plt.ylabel('Quantidade')
plt.title('Número de irmãos por Passageiro')

### Insights

A partir do tratamento e análise da base de dados foi possível observar que a maior parte dos sobreviventes fizeram o tipo de embarque S, este que também continha o maior número de integrantes, 644. Os passageiros do embarque C foram os mais atingidos com o total de 75 mortos de 93.
Ademais. sobre a questão de sexo biológico, a maior parte dos passageiros era homens e  foram os que mais morreram, sendo 468 mortos de um total de 577 homens.

Além disso, foi possível observar a quantidade média de irmãos por passageiros, sendo a maior parte tendo 8 irmãos.

Obs: não consegui plotar os gráficos, pois não está reconheco o matplotlib, pode me ajudar com isso?